src/DiscordHooks.php

```diff
@@ -241,6 +251,13 @@ public static function onUserGroupsChanged( User $user, array $added, array $rem
 	 * @see https://www.mediawiki.org/wiki/Manual:Hooks/UploadComplete
 	 */
 	public static function onUploadComplete( &$image ) {
+		global $wgDiscordNoBots, $wgUser;
+
+		if ( $wgDiscordNoBots && $wgUser->isBot() ) {
+			// Don't continue, this is a bot change
+			return true;
+		}
```

Comment:
Please move this if statement under the `DiscordUtils::isDisabled` one below, and use `$user` instead of `$wgUser`